daikon-logging/logging-event-layout/src/main/java/org/talend/daikon/logging/event/layout/Log4j2JSONLayout.java

```diff
@@ -1,233 +1,257 @@
-     * @param hostInfo
-     * If "true", includes the information about the local host name and IP address.
-     * @param properties
-     * If "true", includes the thread context in the generated JSON.
-     * @param complete
-     * If "true", includes the JSON header and footer, defaults to "false".
-     * @param compact
-     * If "true", does not use end-of-lines and indentation, defaults to "false".
-     * @param eventEol
-     * If "true", forces an EOL after each log event (even if compact is "true"), defaults to "false". This
-     * allows one even per line, even in compact mode.
-     * @param charset
-     * The character set to use, if {@code null}, uses "UTF-8".
-     * @param pairs
-     * MDC attributes
-     * @return A JSON Layout.
-     */
-    @PluginFactory
-    public static AbstractStringLayout createLayout(
-    // @formatter:off
-            @PluginAttribute(value = "locationInfo") final boolean locationInfo,
-            @PluginAttribute(value = "hostInfo", defaultBoolean = true) final boolean hostInfo,
-            @PluginAttribute(value = "properties") final boolean properties,
-            @PluginAttribute(value = "complete") final boolean complete,
-            @PluginAttribute(value = "compact") final boolean compact,
-            @PluginAttribute(value = "eventEol") final boolean eventEol,
-            @PluginAttribute(value = "charset", defaultString = "UTF-8") final Charset charset,
-            @PluginElement("Pairs") final KeyValuePair[] pairs
-            // @formatter:on
-    ) {
-
-        // Unpacke the pairs list
-        final Map<String, String> additionalLogAttributes = unpackPairs(pairs);
-        return new Log4j2JSONLayout(locationInfo, hostInfo, charset, additionalLogAttributes);
+    @Override
+    public String toSerializable(LogEvent event) {
+        final StringBuilder text = toText(event, getStringBuilder(), false);
+        return text.toString();
+    }
 
+    @Override
+    public void encode(LogEvent event, ByteBufferDestination destination) {
+        final StringBuilder text = toText(event, getStringBuilder(), true);
+        final Encoder<StringBuilder> helper = getStringBuilderEncoder();
+        helper.encode(text, destination);
     }
 
-    /**
-     * Formats a {@link org.apache.logging.log4j.core.LogEvent}.
-     *
-     * @param loggingEvent The LogEvent.
-     * @return The JSON representation of the LogEvent.
-     */
     @Override
-    public String toSerializable(final LogEvent loggingEvent) {
-        JSONObject logstashEvent = new JSONObject();
-        JSONObject userFieldsEvent = new JSONObject();
-        HostData host = new HostData();
+    public String getContentType() {
+        return "application/json";
+    }
 
-        // Extract and add fields from log4j2 config, if defined
-        LayoutUtils.addUserFields(additionalAttributes, userFieldsEvent);
+    private StringBuilder toText(LogEvent event, StringBuilder builder, boolean gcFree) {
```

Comment:
```suggestion
    private StringBuilder toText(LogEvent event, StringBuilder builder) {
```
gcFree seems not to be used. (of course the call to this method should also be changed should you access this suggestion.

Reply:
Indeed good catch 👍